pyfr/backends/opencl/blasext.py

```diff
@@ -50,33 +50,45 @@ def errest(self, x, y, z, *, norm):
         if x.traits != y.traits != z.traits:
             raise ValueError('Incompatible matrix types')
 
-        cnt = x.leaddim*x.nrow
-        dtype = x.dtype
+        nrow, ldim, dtype = x.traits
+        ncola, ncolb = x.ioshape[1:]
 
-        # Norm type
-        reduce_expr = 'a + b' if norm == 'l2' else 'max(a, b)'
+        # Reduction workgroup dimensions
+        block = (128, 1, 1)
```

Comment:
File "/home/niki/python3venv/venv/lib/python3.5/site-packages/pyopencl-2016.1-py3.5-linux-x86_64.egg/pyopencl/cffi_cl.py", line 1192, in enqueue_nd_range_kernel
    work_dim = len(global_work_size)
TypeError: object of type 'int' has no len()